arch/arm64/core/cortex_r/arm_mpu.c

```diff
@@ -247,8 +248,13 @@ static int dynamic_areas_init(uintptr_t start, size_t size)
 static int dup_dynamic_regions(struct dynamic_region_info *dst, int len)
 {
 	size_t i;
+	int ret = sys_dyn_regions_num;
 
-	__ASSERT(sys_dyn_regions_num < len, "system dynamic region nums too large.");
+	CHECKIF(!(sys_dyn_regions_num < len)) {
```

Comment:
Would it make it more readable to reverse the polarity of this?  There are several places like this in the PR.  I see that you just changed the __ASSERT to CHECKIF and added the !, but I think it might be more readable as
CHECKIF(sys_dyn_regions_num >= len)

Reply:
I thought about that, but decided not to. It is to keep the core of the conditionals the same as in `__ASSERT()`. So that the brain doesn't have to parse twice, as long as one understands that `__ASSERT() => !CHECKIF()`.